### PR TITLE
#5955 Implement handoverAddress

### PR DESCRIPTION
**Pull request by cristinamghita**
#5955 Display handoverlocation in sales order

### CODE REVIEW

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.swat/de.metas.swat.base/src/main/java/de/metas/adempiere/modelvalidator/Order.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4714908&bid=15746173)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.swat/de.metas.swat.base/src/main/java/de/metas/adempiere/modelvalidator/Order.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4714908&bid=15746173)

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.swat/de.metas.swat.base/src/main/java/de/metas/adempiere/modelvalidator/Order.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4714908&bid=15746173)

File: de.metas.swat/de.metas.swat.base/src/main/java/de/metas/document/impl/DocumentLocationBL.java [hunk 1]
```diff
@@ -186,4 +187,36 @@ private String makeWarehouseAddress(final I_M_Warehouse warehouse)
...
+				|| docHandOverLocation.getHandOver_Partner_ID() <= 0
+				|| docHandOverLocation.getHandOver_Location_ID() <= 0) 
+		{
+			return;
+		}
+
+		final I_C_BPartner bp = InterfaceWrapperHelper.create(docHandOverLocation.getHandOver_Partner(), I_C_BPartner.class);
+
+		
+		final I_C_BPartner_Location bpartnerLocation = InterfaceWrapperHelper.load(docHandOverLocation.getHandOver_Location_ID(), I_C_BPartner_Location.class);
```
Comment (teosarca) [inline de.metas.swat/de.metas.swat.base/src/main/java/de/metas/document/impl/DocumentLocationBL.java]: use BPartnerDAO

File: de.metas.swat/de.metas.swat.base/src/main/java/de/metas/document/impl/DocumentLocationBL.java [hunk 2]
```diff
@@ -186,4 +187,36 @@ private String makeWarehouseAddress(final I_M_Warehouse warehouse)
...
+
+		final de.metas.adempiere.model.I_AD_User user;
+		if (docHandOverLocation.getHandOver_User_ID() > 0)
+		{
+			final I_AD_User userPO = docHandOverLocation.getHandOver_User();
```
Comment (teosarca) [inline de.metas.swat/de.metas.swat.base/src/main/java/de/metas/document/impl/DocumentLocationBL.java]: use BPartnerDAO

File: de.metas.business/src/main/java/de/metas/order/OrderFactory.java [hunk 3]
```diff
@@ -279,6 +279,7 @@ public OrderFactory handOverBPartner(final int bpartnerId, final int bpartnerLoc
 		assertNotBuilt();
 		order.setHandOver_Partner_ID(bpartnerId);
 		order.setHandOver_Location_ID(bpartnerLocationId);
+		order.setHandOver_User_IDD(contactId);
```
Comment (metas-ts) [inline de.metas.business/src/main/java/de/metas/order/OrderFactory.java]: pls remove the suplus `D`

**Comment by metas-ts** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- de.metas.swat/de.metas.swat.base/src/main/java/de/metas/adempiere/modelvalidator/Order.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metasfresh/metasfresh/pullRequest?prid=4714908&bid=15746173)

**Pull request merged**